lib/aws-genai-llm-chatbot-stack.ts

```diff
@@ -100,46 +100,45 @@ export class AwsGenAILLMChatbotStack extends cdk.Stack {
     }
 
     // IDEFICS Interface Construct
-    // This is the model interface recieving messages from the websocket interface via the message topic
+    // This is the model interface receiving messages from the websocket interface via the message topic
     // and interacting with IDEFICS visual language models
     const ideficsModels = models.models.filter(
-      (model) => model.interface === ModelInterface.Idefics
+      (model) => model.interface === ModelInterface.MultiModal
     );
 
     // check if any deployed model requires idefics interface
-    if (ideficsModels.length > 0) {
-      const ideficsInterface = new IdeficsInterface(this, "IdeficsInterface", {
-        shared,
-        config: props.config,
-        messagesTopic: chatBotApi.messagesTopic,
-        sessionsTable: chatBotApi.sessionsTable,
-        byUserIdIndex: chatBotApi.byUserIdIndex,
-        chatbotFilesBucket: chatBotApi.filesBucket,
-      });
 
-      // Route all incoming messages targeted to idefics to the idefics model interface queue
-      chatBotApi.messagesTopic.addSubscription(
-        new subscriptions.SqsSubscription(ideficsInterface.ingestionQueue, {
-          filterPolicyWithMessageBody: {
-            direction: sns.FilterOrPolicy.filter(
-              sns.SubscriptionFilter.stringFilter({
-                allowlist: [Direction.In],
-              })
-            ),
-            modelInterface: sns.FilterOrPolicy.filter(
-              sns.SubscriptionFilter.stringFilter({
-                allowlist: [ModelInterface.Idefics],
-              })
-            ),
-          },
-        })
-      );
+    const ideficsInterface = new IdeficsInterface(this, "IdeficsInterface", {
```

Comment:
Wouldn't be better to also change the construct name to `MultimodalInterface` ?

Reply:
I did that first, but updating an existing stack will take ages, so I decided to keep the old name.